crates/jarscanner/src/lib.rs

```diff
@@ -192,6 +192,29 @@ mod tests {
         let mut vec_expected = set.iter().collect::<Vec<_>>();
         vec_expected.sort();
         assert_eq!(vec_expected, vec!["scala.runtime", "scala.runtime.package"]);
+
+        // We should handle the /$ pattern correctly
+        set.clear();
+        file_name_to_class_names("scala/collection/immutable/$colon$colon$.class", &mut set);
+        let mut vec_expected = set.iter().collect::<Vec<_>>();
+        vec_expected.sort();
+        assert_eq!(
+            vec_expected,
+            vec!["scala.collection.immutable.colon.colon",]
```

Comment:
this doesn't seem quite right either and brings up a design flaw in the current system: in scala the name of this symbol would be `::` but that's not how it gets encoded in the jar, or for java... so java could refer to it as `$colon$colon` but scala wouldn't.

We don't need to fix this here, but it is an issue with the system: we need to translate symbol names to some universal encoding, but that's not totally obvious.